drivers/flash/flash_stm32_ospi.c

```diff
@@ -329,7 +368,17 @@ static int stm32_ospi_mem_ready(OSPI_HandleTypeDef *hospi, uint8_t nor_mode, uin
 	s_command.Address = 0U;
 
 	if (nor_mode == OSPI_OPI_MODE) {
-		s_command.DummyCycles = (nor_rate == OSPI_DTR_TRANSFER)
+		s_command.DummyCycles = (nor_rate == OSPI_DTR_TRANSFER);
+		s_command.Instruction = (nor_mode == OSPI_SPI_MODE)
+				? SPI_NOR_CMD_RDSR
+				: SPI_NOR_OCMD_RDSR;
+		s_command.AddressMode = (nor_mode == OSPI_SPI_MODE)
+				? HAL_OSPI_ADDRESS_NONE
+				: HAL_OSPI_ADDRESS_8_LINES;
```

Comment:
This piece of code is inside `if (nor_mode == OSPI_OPI_MODE) {`, so result of those comparison is known.

Reply:
no more applicable